### PR TITLE
Use new syntax to set environment variable

### PR DESCRIPTION
**Pull request by martincostello**
Use new syntax to set environment variable value.

See https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
         $QuarterHours = [Math]::Floor($Now.Minute / 15.0)
         $Revision = $Hours + $QuarterHours + 1
         $BuildId = $Now.ToString("yyyyMMdd") + "." + $Revision
-        Write-Host "::set-env name=_AspNetContribBuildNumber::${BuildId}"
+        Write-Host "_AspNetContribBuildNumber=${BuildId}" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```
Comment (kevinchalet) [inline .github/workflows/build.yml]: I'm definitely not a PowerShell expert so I may be wrong, but doesn't `Write-Host` directly write to the console output?
Instead, shouldn't we do something like:

```
"_AspNetContribBuildNumber=${BuildId}" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```

Or:

```
Add-Content -Path $env:GITHUB_PATH -Encoding utf8 -Value "_AspNetContribBuildNumber=${BuildId}"
```

Comment (martincostello) [reply .github/workflows/build.yml]: ¯\_(ツ)_/¯

I just copied this and replaced the `echo` alias: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

Comment (kevinchalet) [reply .github/workflows/build.yml]: I suspect the two syntaxes don't behave exactly the same way in this regard:

image

Comment (martincostello) [reply .github/workflows/build.yml]: Looks like I just mixed up what `echo` was the alias for: https://stackoverflow.com/a/17623887/1064169

**Comment by martincostello** [comment]
Hmm...

I'm not sure that actually worked correctly. The revision is coming out as `1` even though the code that calculates the number hasn't changed and it seeme to work as expected if I run it locally:

```powershell
C:\Coding> $Now = (Get-Date).ToUniversalTime()
C:\Coding> $Hours = $Now.Hour * 4
C:\Coding> $QuarterHours = [Math]::Floor($Now.Minute / 15.0)
C:\Coding> $Revision = $Hours + $QuarterHours + 1
C:\Coding> $BuildId = $Now.ToString("yyyyMMdd") + "." + $Revision
C:\Coding> Write-Output "_AspNetContribBuildNumber=${BuildId}"
_AspNetContribBuildNumber=20201109.90
```

**Comment by kevinchalet** [comment]
> I'm not sure that actually worked correctly.

Is that because you used `$env:GITHUB_PATH` instead of `$env:GITHUB_ENV`?

**Pull request merged**